src/api/app/controllers/webui/project_controller.rb

```diff
@@ -95,6 +95,8 @@ def subprojects
   def new
     @project = Project.new
     @project.name = params[:name] if params[:name]
+
+    @show_restore_message = params[:restore_option] && Project.deleted?(params[:name])
```

Comment:
It's set here: https://github.com/openSUSE/open-build-service/pull/3271/commits/75147a84df177f4a920373bc0e0e3867d7ed78ad#diff-86772a8e6d058f9f138b75ac35927ffdR306

Imagine the following scenario: A user is about to create a project, first it checks if the name was taken in the past. If so, the user gets redirected to the creation page with the parameter `?restore_option=true` -> the message is shown.